sensor/common/store/resolver/resolver.go

```diff
@@ -49,3 +50,10 @@ func ResolveAllDeployments() DeploymentReference {
 		return ids
 	}
 }
+
+// ResolveDeploymentsByImages returns a function that returns a list of deployment ids based on a slice of images
+func ResolveDeploymentsByImages(images ...imagecacheutils.CacheKeyProvider) DeploymentReference {
```

Comment:
Nit: I'm a little uneasy with using `imagecacheutils.CacheKeyProvider` as the abstraction for the resolver and store functions. Here's my reasoning:

`imagecacheutils.CacheKeyProvider` surfaces an _internal_ design decision of the resolver function: we are using `imagecacheutils` functions to compare the image entries. On the other hand, `storage.*` type object are the canonical model of all ACS entities. Which I think is a much better "common" interface for communicating deployment resolution. Additionally, this ties the method interface to `imagecacheutils`. If in the future we want to index deployment differently using another property of `storage.Image`, we'd have to change the interface or the method signatures. 

With that said, I would prefer to use `storage.Image` in the resolver and store methods.

Reply:
I had `storage.Image` at first and I changed it to that interface to do it the “go way” (maybe it would’ve been better to define a new interface in the store package). I agree with your reasoning though and since we are using storage structures everywhere else I’ll change it back to `storage.Image`.